common/src/main/scala/org/apache/celeborn/common/util/Utils.scala

```diff
@@ -245,8 +245,15 @@ object Utils extends Logging {
     }
   }
 
-  def selectRandomPort(from: Int, to: Int): Int = {
-    ScalaRandom.nextInt(to - from) + from
+  /**
+   * Selects a random port within the specified range.
+   *
+   * @param from the lower bound of the range (inclusive)
+   * @param until the upper bound of the range (exclusive)
+   * @return a randomly selected port within the range [from, until)
+   */
+  def selectRandomPort(from: Int, until: Int): Int = {
+    ScalaRandom.nextInt(until - 1 - from) + from
```

Comment:
<img width="1141" alt="image" src="https://github.com/user-attachments/assets/e9aabad8-64db-47d6-b464-993eaa3a1936">

Due this method is only used by UT, both for scala and java UTs.

To minimize the code change, I just change the `selectRandomPort`  method to make the upper bound be exclusive.
cc @pan3793